main.py

```diff
@@ -0,0 +1,203 @@
+    for update in updates["result"]:
+        update_ids.append(int(update["update_id"]))
+    return max(update_ids)
+
+def send_message(text, chat_id):
+    text = urllib.parse.quote_plus(text)
+    url = URL + "sendMessage?text={}&chat_id={}".format(text, chat_id)
+    get_url(url)
+
+def get_new_emails(imap_login, imap_password):
+    ix = imap_login.index('@')
+    EMAIL = imap_login
+    PASSWORD = imap_password
+    SERVER = 'imap.' + imap_login[ix + 1:]
+    if imap_login[ix + 1:] == 'bk.ru':
+        SERVER = 'imap.mail.ru'
+    elif imap_login[ix + 1:] == 'phystech.edu':
+        SERVER = 'imap.gmail.com'
+    mail = imaplib.IMAP4_SSL(SERVER)
+    mail.login(EMAIL, PASSWORD)
+    mail.select('inbox', readonly=False)
+
+    status, data = mail.search(None, 'UNSEEN')
+    mail_ids = []
+    for block in data:
+        mail_ids += block.split()
+
+    result = []
+
+    for i in mail_ids:
+        status, data = mail.fetch(i, '(RFC822)')
+
+        for response_part in data:
+            if isinstance(response_part, tuple):
+                message = email.message_from_bytes(response_part[1])
+                typ, data = mail.store(i, '+FLAGS', '\\Seen')
+
+                from_ = email.header.decode_header(message['from'])[0]
+                mail_from = from_[0].decode(from_[1], 'replace') \
+                    if from_[1] else from_[0]
+                subject_ = email.header.decode_header(message['subject'])[0]
+                mail_subject = subject_[0].decode(subject_[1], 'replace') \
+                    if subject_[1] else subject_[0]
+
+                if message.is_multipart():
+                    mail_content = ''
+                    for part in message.get_payload():
+                        if part.get_content_type() == 'text/plain':
+                            payload_bytes = part.get_payload(decode=True)
+                            charset = part.get_content_charset('utf-8')
+                            mail_content += payload_bytes\
+                                .decode(charset, 'replace')
+                else:
+                    charset = message.get_charset('utf-8')
+                    mail_content = message.get_payload(decode=True)\
+                        .decode(charset, 'replace')
+
+
+                result += [{'from': mail_from, 'subj': mail_subject,
+                            'content': mail_content}]
+    mail.close()
+    return result
+
+@orm.db_session()
+def handle_updates(grouped_updates):
+    for chat_id, g_upd in grouped_updates.items():
+        current_chat = Chats.get(chat_id=chat_id)
+        current_state = current_chat.state if current_chat else 0
+        current_login = current_chat.login if current_chat else None
+        current_passwd = current_chat.passwd if current_chat else None
+        for upd in g_upd:
+            text = upd["message"]["text"]
+            if text == "/start" and current_state == 0:
+                start_message = """Welcome to Emails2Telegram bot!
```

Comment:
Maybe it would be better to store bot messages in separate variables
(At least, it would be easier to change them)